components/user_settings/modal/user_settings_modal.tsx

```diff
@@ -91,10 +107,10 @@ class UserSettingsModal extends React.Component {
         this.modalBodyRef = React.createRef();
     }
 
-    handleResend = (email) => {
+    handleResend = (email: string) => {
         this.setState({resendStatus: 'sending'});
 
-        this.props.actions.sendVerificationEmail(email).then(({data, error: err}) => {
+        this.props.actions.sendVerificationEmail(email).toPromise().then(({data, err}: {data: any; err: any}) => {
```

Comment:
This shouldn't need explicit converting, you should be able to have the return type of `sendVerificationEmail` return `Promise<ActionResult>`